packages/notifier/src/notifier.js

```diff
@@ -1,32 +1,29 @@
 /// <reference types="ses"/>
 
 import { assert } from '@agoric/assert';
-import { makePromiseKit } from '@endo/promise-kit';
-import { E } from '@endo/eventual-send';
-import { Far } from '@endo/marshal';
-import { makeAsyncIterableFromNotifier } from './asyncIterableAdaptor.js';
+import { E, Far } from '@endo/far';
 
 import './types-ambient.js';
+import { makePublishKit } from './publish-kit.js';
+import { subscribeLatest } from '../tools/subscribe.js';
 
 /**
  * @template T
  * @param {ERef<BaseNotifier<T> | NotifierInternals<T>>} sharableInternalsP
  * @returns {AsyncIterable<T> & SharableNotifier<T>}
  */
 export const makeNotifier = sharableInternalsP => {
-  const asyncIterable = makeAsyncIterableFromNotifier(sharableInternalsP);
-
   /** @type {AsyncIterable<T> & SharableNotifier<T>} */
   const notifier = Far('notifier', {
-    ...asyncIterable,
+    ...subscribeLatest(sharableInternalsP),
 
     /**
      * Use this to distribute a Notifier efficiently over the network,
      * by obtaining this from the Notifier to be replicated, and applying
      * `makeNotifier` to it at the new site to get an equivalent local
      * Notifier at that site.
      */
-    getSharableNotifierInternals: () => sharableInternalsP,
+    getSharableNotifierInternals: async () => sharableInternalsP,
```

Comment:
looks lke the `async` is to force the return type to be Promise. the highlights that `sharableInternalsP` may not be a promise. Consider `sharableInternalsE` or `sharableInternalsRef`